Source/Evolve/WarpXEvolve.cpp

```diff
@@ -136,7 +136,39 @@ WarpX::Evolve (int numsteps)
                             *Efield_aux[lev][0],*Efield_aux[lev][1],*Efield_aux[lev][2],
                             *Bfield_aux[lev][0],*Bfield_aux[lev][1],*Bfield_aux[lev][2]);
             }
+
+            // The hybrid-PIC algorithm uses the charge and current density
+            // from both the current and previous step when updating the
+            // fields, so we deposit the ion charge and current now in the
+            // temp multifab locations.
+            if (electromagnetic_solver_id == ElectromagneticSolverAlgo::HybridPIC)
```

Comment:
As discussed offline, this relies on `is_synchronized`, but `is_synchronized` may not always correspond to the very first timestep. It may be better to instead check `step == step_begin`.